powersimdata/scenario/create.py

```diff
@@ -49,8 +56,24 @@ def __init__(self, scenario):
                 ("engine", ""),
             ]
         )
+        self.exported_methods = set(default_exported_methods)
         super().__init__(scenario)
 
+    def __getattr__(self, name):
```

Comment:
Does this fail when called with a method that is not in the builder now? e.g. the first if statement in `Scenario.__getattr__`

Reply:
```
>>> import powersimdata as psd
>>> scenario = psd.Scenario()
>>> scenario.foo
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\DanielOlsen\repos\bes\PowerSimData\powersimdata\scenario\scenario.py", line 49, in __getattr__
    return self.state._custom_getattr_error_message(name)
  File "C:\Users\DanielOlsen\repos\bes\PowerSimData\powersimdata\scenario\create.py", line 75, in _custom_getattr_error_message
    return self.__getattr__(name)
  File "C:\Users\DanielOlsen\repos\bes\PowerSimData\powersimdata\scenario\create.py", line 69, in __getattr__
    raise AttributeError(
AttributeError: Create object without a builder set has no attribute foo. Did you forget to run set_builder?
>>> scenario.set_builder(interconnect="Texas")
Transferring ScenarioList.csv from server
100%|#######################################| 543k/543k [00:00<00:00, 4.18Mb/s]
Reading bus.csv
Reading plant.csv
Reading gencost.csv
Reading branch.csv
Reading dcline.csv
Reading sub.csv
Reading bus2sub.csv
Reading zone.csv
--> Summary
# Existing study
test | base | Anchor | Julia | Texas_Crisis
# Available profiles
demand: vJan2021
hydro: vJan2021
solar: vJan2021
wind: vJan2021
>>> scenario.foo
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\DanielOlsen\repos\bes\PowerSimData\powersimdata\scenario\scenario.py", line 49, in __getattr__
    return self.state._custom_getattr_error_message(name)
  File "C:\Users\DanielOlsen\repos\bes\PowerSimData\powersimdata\scenario\create.py", line 75, in _custom_getattr_error_message
    return self.__getattr__(name)
  File "C:\Users\DanielOlsen\repos\bes\PowerSimData\powersimdata\scenario\create.py", line 67, in __getattr__
    raise AttributeError(f"Create object has no attribute {name}")
AttributeError: Create object has no attribute foo
```

Reply:
Thanks, TIL the difference between `getattr` and `__getattr__`